pkg/registry/servicecatalog/broker/strategy.go

```diff
@@ -0,0 +1,129 @@
+
+// this was copied from where else and edited to fit our objects
+
+import (
+	kapi "k8s.io/kubernetes/pkg/api"
+	"k8s.io/kubernetes/pkg/runtime"
+	"k8s.io/kubernetes/pkg/util/validation/field"
+
+	sc "github.com/kubernetes-incubator/service-catalog/pkg/apis/servicecatalog"
+)
+
+/* Begin Create Definition */
+
+type brokerCreateStrategy struct {
+	runtime.ObjectTyper // inherit ObjectKinds method
+	kapi.NameGenerator  // GenerateName method for CreateStrategy
+}
+
+// implements RESTCreateStrategy interface
+var createStrategy = brokerCreateStrategy{
+	// embeds to pull in existing code behavior from upstream
+
+	// this has an interesting NOTE on it. Not sure if it applies to us.
+	ObjectTyper: kapi.Scheme,
+	// use the generator from upstream k8s, or implement method
+	// `GenerateName(base string) string`
+	NameGenerator: kapi.SimpleNameGenerator,
+}
+
+// Canonicalize is called after validate. What happens if it creates
+// an object to persist that does not pass validate? (I think the
+// answer is "don't do that"). Frequently is an empty method or a type
+// check. May mutate the object.
+func (brokerCreateStrategy) Canonicalize(obj runtime.Object) {}
+
+// Are brokers namespace scoped?
+func (brokerCreateStrategy) NamespaceScoped() bool {
+	return false
+}
+
+// PrepareForCreate recieves
+func (brokerCreateStrategy) PrepareForCreate(ctx kapi.Context, obj runtime.Object) {
+	// coerce to our specific object type. (Should we type check?)
+	broker := obj.(*sc.Broker)
+	// Is there anything to pull out of the context `ctx`?
+
+	// Creating a brand new object, thus it must have no status.
+	// Set to "creating" ?
+	broker.Status = sc.BrokerStatus{}
+	broker.Status.Conditions = []sc.BrokerCondition{}
+}
+
+func (brokerCreateStrategy) Validate(ctx kapi.Context, obj runtime.Object) field.ErrorList {
+	return validateBroker(obj.(*sc.Broker))
+}
+
+/* End Create Definition */
+
+/* Begin Delete Definition */
+// implements RESTDeleteStrategy interface
+type brokerDeleteStrategy struct {
+	runtime.ObjectTyper // inherit ObjectKinds method
+}
+
+// Strategy implements
+var deleteStrategy = brokerDeleteStrategy{
+	// this has an interesting NOTE on it. Not sure if it applies to us.
+	ObjectTyper: kapi.Scheme,
+}
+
+type brokerUpdateStrategy struct {
+	runtime.ObjectTyper // inherit ObjectKinds method
+}
+
+// There is no implementation code for Delete.
+/* End Delete Definition */
+
+/* Begin Update Definition */
+
+// implements RESTUpdateStrategy interface
+var updateStrategy = brokerUpdateStrategy{
+	// this has an interesting NOTE on it. Not sure if it applies to us.
+	ObjectTyper: kapi.Scheme,
+}
+
+func (brokerUpdateStrategy) AllowCreateOnUpdate() bool {
+	return false
+}
+
+func (brokerUpdateStrategy) AllowUnconditionalUpdate() bool {
+	return false
+}
+
+func (brokerUpdateStrategy) Canonicalize(obj runtime.Object) {}
+
+// Are brokers namespace scoped? Is the namespace concerned about an
+// external namespace or some storage namespace?
```

Comment:
I've been wondering the same thing throughout this code. The 'store' object seems to be less storage specific and more of an amalgamated translation layer. It receives base unversioned api objects incoming from the http layer, and validates and munges them into something storable. It does not actually do storage. It's more a facade in front of the real storage.